renew/mail.go

```diff
@@ -0,0 +1,105 @@
+package renew
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/wopta/goworkspace/lib"
+	"github.com/wopta/goworkspace/mail"
+	"github.com/wopta/goworkspace/models"
+)
+
+type RenewReq struct {
+	Date            string `json:"date"`
+	DaysBeforeRenew string `json:"days_before_renew"`
+}
+
+func RenewMailFx(w http.ResponseWriter, r *http.Request) (string, interface{}, error) {
+	var (
+		err             error
+		date            = time.Now().UTC()
+		daysBeforeRenew = 10
+		query           bytes.Buffer
+		params          = make(map[string]interface{})
+		req             RenewReq
+	)
+
+	log.SetPrefix("[RenewMailFx] ")
+	defer func() {
+		collectionPrefix = ""
+		if err != nil {
+			log.Printf("error: %s", err.Error())
+		}
+		log.Println("Handler end -------------------------------------------------")
+	}()
+	log.Println("Handler start -----------------------------------------------")
+
+	body := lib.ErrorByte(io.ReadAll(r.Body))
+	defer r.Body.Close()
+	err = json.Unmarshal(body, &req)
+	if err != nil {
+		return "", nil, err
+	}
+
+	if req.Date != "" {
+		tmpDate, err := time.Parse(time.DateOnly, req.Date)
+		if err != nil {
+			log.Printf("error parsing request date: %s", err.Error())
+			return "", nil, err
+		}
+		date = tmpDate
+	}
+
+	if req.DaysBeforeRenew != "" {
+		tmpDays, err := strconv.Atoi(req.DaysBeforeRenew)
+		if err != nil {
+			log.Printf("error parsing target date: %s", err.Error())
+			return "", nil, err
+		}
+		daysBeforeRenew = tmpDays
+	}
+
+	targetDate := date.AddDate(0, 0, daysBeforeRenew)
+	log.Printf("Date is: %v; targetDate is: %v", date, targetDate)
+
+	params["isRenewable"] = true
+	params["channel"] = models.ECommerceChannel
+	params["targetYear"] = int64(targetDate.Year())
+	params["targetMonth"] = int64(targetDate.Month())
+	params["targetDay"] = int64(targetDate.Day())
```

Comment:
Here you should add also:

- `isDeleted = false`
- `isPay = false`

So that we are sure first to send email only to people for which renew is in progress and payment has not been made. It could happen that a node from the network contacts the customer before -10 days and proceed with the payment. In this case we don't want to send an email saying you can pay .....